src/NuGet.Core/NuGet.Configuration/PackageSource/PackageSourceProvider.cs

```diff
@@ -103,20 +107,20 @@ private static IEnumerable<PackageSource> LoadConfigurationDefaultSources(IEnume
                     var isEnabled = !disabledSources.Contains(name);
                     var packageSource = ReadPackageSource(setting, isEnabled, settings);
 
-                    packageIndex = AddOrUpdateIndexedSource(packageSourceLookup, packageIndex, packageSource, byName ? packageSource.Name : packageSource.Source);
+                    packageIndex = AddOrUpdateIndexedSource(packageSourceLookup, packageIndex, packageSource);
                 }
-            }
-            return packageSourceLookup;
-        }
 
-        private static Dictionary<string, IndexedPackageSource> LoadPackageSourceLookupByName(ISettings settings)
-        {
-            return LoadPackageSourceLookup(byName: true, settings);
-        }
+                packageSources = new(capacity: packageSourceLookup.Count);
+                packageSources.AddRange(packageSourceLookup.Values
+                    .OrderBy(psi => psi.Index).
+                    Select(psi => psi.PackageSource));
+            }
+            else
+            {
+                packageSources = new List<PackageSource>();
```

Comment:
Can we use `Enumerable.Empty<PackageSource>` here?

Reply:
The default constructor already creates a list with zero capacity.

.NET Framework: https://referencesource.microsoft.com/#mscorlib/system/collections/generic/list.cs,49

.NET (Core): https://github.com/dotnet/runtime/blob/cbc501ca196371572c38f8d12a66969864d99c08/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/List.cs#L33-L40

Reply:
I was under the impression that default constructor creates an array of with some default capacity. Thank you for the clarification.